lib/ansible/modules/cloud/amazon/cloudfront_distribution.py

```diff
@@ -0,0 +1,2106 @@
+                    msg="origins[].domain_name must be specified for an origin")
+            if 'id' not in origin:
+                origin['id'] = self.__default_datetime_string
+            if 'custom_headers' in origin and streaming:
+                self.module.fail_json(
+                    msg="custom_headers has been specified for a streaming " +
+                    "distribution. custom headers are for " +
+                    "web distributions only")
+            if ('custom_headers' in origin and
+               len(origin.get('custom_headers')) > 0):
+                for custom_header in origin.get('custom_headers'):
+                    if('header_name' not in custom_header or
+                       'header_value' not in custom_header):
+                        self.module.fail_json(
+                            msg="both origins[].custom_headers.header_name " +
+                            "and origins[].custom_headers.header_value must " +
+                            "be specified")
+                origin['custom_headers'] = helpers.python_list_to_aws_list(
+                    origin.get('custom_headers'))
+            else:
+                origin['custom_headers'] = helpers.python_list_to_aws_list()
+            if self.__s3_bucket_domain_identifier in origin.get('domain_name'):
+                if('s3_origin_config' not in origin or
+                   'origin_access_identity' not in origin.get(
+                        's3_origin_config')):
+                    origin["s3_origin_config"] = {}
+                    origin['s3_origin_config']['origin_access_identity'] = (
+                        default_s3_origin_access_identity or '')
+            else:
+                if 'custom_origin_config' not in origin:
+                    origin['custom_origin_config'] = {}
+                custom_origin_config = origin.get('custom_origin_config')
+                if 'origin_protocol_policy' not in custom_origin_config:
+                    custom_origin_config['origin_protocol_policy'] = (
+                        self.__default_custom_origin_protocol_policy)
+                else:
+                    self.validate_attribute_with_allowed_values(
+                        custom_origin_config.get('origin_protocol_policy'),
+                        'origins[].custom_origin_config.origin_protocol_policy',
+                        self.__valid_origin_protocol_policies)
+                if 'origin_read_timeout' not in custom_origin_config:
+                    custom_origin_config['origin_read_timeout'] = (
+                        self.__default_custom_origin_read_timeout)
+                if 'origin_keepalive_timeout' not in custom_origin_config:
+                    custom_origin_config['origin_keepalive_timeout'] = (
+                        self.__default_custom_origin_keepalive_timeout)
+                if 'http_port' not in custom_origin_config:
+                    custom_origin_config['h_t_t_p_port'] = (
```

Comment:
fixed for all occurrences, thought that the pep8 limit was 79 chars for some unknown reason